modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1136,6 +1138,10 @@ def add_col_field(table, col_name, table_col_name):
                     and (not pyarrow.types.is_null(field.type))
                     and (
                         pyarrow.types.is_string(field.type)
+                        or (
+                            (field.type.bit_width == cur_field.type.bit_width)
+                            and pyarrow.types.is_floating(field.type)
+                        )
```

Comment:
is it possible to split this huge condition into several ones? It's really hard to read it.

Maybe something like:
```python
is_cur_none = cur_field is None or pyarrow.types.is_null(cur_field.type)

allow_upcast = not pyarrow.types.is_string(
    cur_field.type
) and not pyarrow.types.is_null(field.type)

can_upcast_to_float = (
    field.type.bit_width == cur_field.type.bit_width
    and pyarrow.types.is_floating(field.type)
)
can_upcast_to_string = pyarrow.types.is_string(field.type)
will_not_lose_precision = field.type.bit_width > cur_field.type.bit_width

if is_cur_none or (
    allow_upcast
    and (can_upcast_to_string or can_upcast_to_float or will_not_lose_precision)
):
    col_fields[key] = field

```

Reply:
Will try to rephrase. Your example fails, at least, if cur_field is none, or some field is string. The conditions depend on each other.

Reply:
Done.